runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPanel.java

```diff
@@ -658,4 +467,112 @@ public void focusLost(FocusEvent e)
 		revalidate();
 		getScrollPane().getVerticalScrollBar().setValue(0);
 	}
+
+	private void changeConfiguration(Config config, JComponent component, ConfigDescriptor cd, ConfigItemDescriptor cid)
+	{
+		ConfigItem configItem = cid.getItem();
+
+		if (!Strings.isNullOrEmpty(configItem.warning()))
+		{
+			final int result = JOptionPane.showOptionDialog(component, configItem.warning(),
+				"Are you sure?", JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE,
+				null, new String[]{"Yes", "No"}, "No");
+
+			if (result != JOptionPane.YES_OPTION)
+			{
+				openGroupConfigPanel(config, cd);
+				return;
+			}
+		}
+
+		if (component instanceof JCheckBox)
```

Comment:
This can also be just switch. E.g switch component.getClass()

Reply:
I don't think this is possible in Java, unless you mean to use the name of the class, which is possible but probably not a good idea?

Reply:
Ah right, forgot that Java is bad. Alright :d At least make it chain of if elseif